p2p/src/network/kad/bootstrap/p2p_network_kad_bootstrap_reducer.rs

```diff
@@ -42,20 +44,26 @@ fn prepare_next_request(
 }
 
 impl P2pNetworkKadBootstrapState {
-    pub fn reducer(
-        &mut self,
-        routing_table: &P2pNetworkKadRoutingTable,
+    pub fn reducer<Action, State>(
+        mut state_context: Substate<Action, State, Self>,
         action: ActionWithMeta<&P2pNetworkKadBootstrapAction>,
         filter_addrs: bool,
-    ) -> Result<(), String> {
+    ) -> Result<(), String>
+    where
+        State: SubstateAccess<Self> + SubstateAccess<P2pState> + SubstateAccess<P2pNetworkKadState>,
+        Action: crate::P2pActionTrait<State>,
+    {
         let (action, meta) = action.split();
+        let discovery_state: &P2pNetworkKadState = state_context.get_state().substate()?;
+        let routing_table = &discovery_state.routing_table;
```

Comment:
@0xMimir something we were discussing with @dkuehr is that `get_state()` shouldn't be used (it kind of violates the phase separation between state updates and control flow, I see it was introduced for some testing needs, but it was not part of the original API because of I just mentioned).

It would be better to pass these routing table from the caller here, would that complicate things too much?